src/components/ConnectionConfig.tsx

```diff
@@ -0,0 +1,313 @@
+import { awsAuthProviderOptions } from '../providers';
+import { css } from '@emotion/css';
+import { NewConnectionConfig } from './NewConnectionConfig';
+
+export const DEFAULT_LABEL_WIDTH = 28;
+const DS_TYPES_THAT_SUPPORT_TEMP_CREDS = ['cloudwatch'];
+const toOption = (value: string) => ({ value, label: value });
+const isAwsAuthType = (value: any): value is AwsAuthType => {
+  return typeof value === 'string' && awsAuthProviderOptions.some((opt) => opt.value === value);
+};
+export interface ConnectionConfigProps<
+  J extends AwsAuthDataSourceJsonData = AwsAuthDataSourceJsonData,
+  S = AwsAuthDataSourceSecureJsonData
+> extends DataSourcePluginOptionsEditorProps<J, S> {
+  standardRegions?: string[];
+  loadRegions?: () => Promise<string[]>;
+  defaultEndpoint?: string;
+  skipHeader?: boolean;
+  skipEndpoint?: boolean;
+  children?: React.ReactNode;
+  labelWidth?: number;
+  inExperimentalAuthComponent?: boolean;
+  externalId?: string;
+  newFormStylingEnabled?: boolean;
+}
+
+export const ConnectionConfig: FC<ConnectionConfigProps> = (props: ConnectionConfigProps) => {
+  const [isARNInstructionsOpen, setIsARNInstructionsOpen] = useState(false);
+  const [regions, setRegions] = useState((props.standardRegions || standardRegions).map(toOption));
+  const { loadRegions, onOptionsChange, skipHeader = false, skipEndpoint = false } = props;
+  const { labelWidth = DEFAULT_LABEL_WIDTH, options, inExperimentalAuthComponent } = props;
+  let profile = options.jsonData.profile;
+  if (profile === undefined) {
+    profile = options.database;
+  }
+  const tempCredsFeatureEnabled =
+    config.featureToggles.awsDatasourcesTempCredentials && DS_TYPES_THAT_SUPPORT_TEMP_CREDS.includes(options.type);
+  const awsAssumeRoleEnabled = config.awsAssumeRoleEnabled ?? true;
+  const awsAllowedAuthProviders = useMemo(
+    () =>
+      config.awsAllowedAuthProviders
+        .filter((provider) => (provider === AwsAuthType.GrafanaAssumeRole ? tempCredsFeatureEnabled : true))
+        .filter(isAwsAuthType),
+    [tempCredsFeatureEnabled]
+  );
+  const currentProvider = awsAuthProviderOptions.find((p) => p.value === options.jsonData.authType);
+
+  useEffect(() => {
+    // Make sure a authType exists in the current model
+    if (!currentProvider && awsAllowedAuthProviders.length) {
+      onOptionsChange({
+        ...options,
+        jsonData: {
+          ...options.jsonData,
+          authType: awsAllowedAuthProviders[0],
+        },
+      });
+    }
+  }, [currentProvider, options, onOptionsChange, awsAllowedAuthProviders]);
+
+  useEffect(() => {
+    if (!loadRegions) {
+      return;
+    }
+
+    loadRegions().then((regions) => setRegions(regions.map(toOption)));
+  }, [loadRegions]);
+
+  const inputWidth = inExperimentalAuthComponent ? 'width-20' : 'width-30';
+  const styles = useStyles2(getStyles);
+
+  return (
+    <>
+      {props.newFormStylingEnabled ? (
+        <NewConnectionConfig
```

Comment:
I'm getting into extremely nitpicky territory here so please don't feel the need to change the code, but it's been something I've been thinking about with my own code so I thought maybe I'd share:

It can sometimes be nice to think about adding changes in such a way that we are minimizing future changes, so as to feel more confidence when we make them and reduce cleanup. 

For example whenever we end up removing this feature flag we're going to have lingering references to an unused prop `newFormStylingEnabled` and we're going to have a component that does nothing except render a component (so we'll probably have to remove the NewConnectionConfig file and put it back in here). 

Another alternative could be to:
- directly import the feature flag here (then you don't need to pass the feature flag prop at all)
- Keep the new ConnectionConfig in the file called ConnectionConfig, and put all the old stuff in a new file OldConnectionConfig, so that eventually you just delete that file.

Reply:
I agree! I tried to do that in the rest of the files that I hadn't done yet, but I kept this one, as the change to move to OldConnectionConfig now, will take more than putting things from NewConnectionConfig back. I don't think it's super worth it for this specific one right now